### PR TITLE
(maint) update setup struct from lite engine to internal response struct

### PR DESCRIPTION
**Pull request by eoinmcafee00**

### CODE REVIEW

File: command/delegate/delegate.go [hunk 1]
```diff
@@ -30,12 +30,18 @@ import (
...
-	poolManager         *vmpool.Manager
-}
+type (
+	delegateCommand struct {
+		envfile             string
+		awsPoolfile         string
+		defaultPoolSettings vmpool.DefaultSettings
+		poolManager         *vmpool.Manager
+	}
+	setupResponse struct {
```
Comment (tphoney) [inline command/delegate/delegate.go:40]: can you put the setupResponse in https://github.com/drone-runners/drone-runner-aws/blob/master/command/delegate/data.go

File: command/delegate/delegate.go [hunk 2]
```diff
@@ -303,12 +309,12 @@ func (c *delegateCommand) handleSetup() http.HandlerFunc {
 			w.WriteHeader(http.StatusInternalServerError)
 			return
 		}
-		JSON(w, api.SetupResponse{IPAddress: instance.IP, InstanceID: instance.ID}, http.StatusOK)
+		JSON(w, setupResponse{IPAddress: instance.IP, InstanceID: instance.ID}, http.StatusOK)
```
Comment (marko-gacesa) [inline command/delegate/delegate.go:312]:
```suggestion
		JSON(w, struct {
			InstanceID string `json:"instance_id,omitempty"`
			IPAddress  string `json:"ip_address,omitempty"`
		}{
			IPAddress: instance.IP,
			InstanceID: instance.ID
		}, http.StatusOK)
```

File: command/delegate/delegate.go [hunk 3]
```diff
@@ -303,12 +309,12 @@ func (c *delegateCommand) handleSetup() http.HandlerFunc {
...
 		logrus.
 			WithField("pool", poolName).
 			WithField("ip", instance.IP).
 			WithField("id", instance.ID).
-			WithField("response", *setupResponse).
+			WithField("response", *liteEngineSetupResponse).
```
Comment (tphoney) [inline command/delegate/delegate.go:317]: this should stay WithField("response", *setupResponse).

**Review by tphoney — CHANGES_REQUESTED** [review]

File: command/delegate/delegate.go (hunk 2 above)
Comment (marko-gacesa) [reply command/delegate/delegate.go:312]: Because the struct does not have to exist outside of this function, this one place.

**Pull request closed without merging**